cl/search/management/commands/cl_index_parent_and_child_docs.py

```diff
@@ -146,95 +330,228 @@ def handle(self, *args, **options):
-                    ).set(queue=queue).apply_async()
-                else:
-                    index_parent_or_child_docs.si(
-                        chunk,
-                        search_type,
-                        document_type,
-                        testing_mode=testing_mode,
-                    ).set(queue=queue).apply_async()
+
+                match task_to_use:
+                    case "index_parent_and_child_docs":
+                        index_parent_and_child_docs.si(
+                            chunk, search_type, testing_mode=testing_mode
+                        ).set(queue=queue).apply_async()
+
+                    case "index_parent_or_child_docs":
+                        index_parent_or_child_docs.si(
+                            chunk,
+                            search_type,
+                            document_type,
+                            testing_mode=testing_mode,
+                        ).set(queue=queue).apply_async()
+                    case "remove_parent_and_child_docs_by_query":
+                        remove_parent_and_child_docs_by_query.si(
+                            ESRECAPDocument.__name__, chunk, event_doc_type
+                        ).set(queue=queue).apply_async()
+                    case "update_children_docs_by_query":
+                        update_children_docs_by_query.si(
+                            ESRECAPDocument.__name__,
+                            item_id,
+                            changed_fields,
+                            fields_map,
+                            event_doc_type,
+                        ).set(queue=queue).apply_async()
 
                 chunk = []
+
                 self.stdout.write(
-                    "\rProcessed {}/{}, ({:.0%}), last PK indexed: {},".format(
+                    "\rProcessed {}/{}, ({:.0%}), last {} PK indexed: {},".format(
                         processed_count,
                         count,
                         processed_count * 1.0 / count,
+                        event_doc_type.value if event_doc_type else "",
                         item_id,
                     )
                 )
-            if not processed_count % 1000:
-                # Log every 1000 parent documents processed.
-                log_last_document_indexed(
-                    item_id, compose_redis_key(search_type)
-                )
-        self.stdout.write(
-            f"Successfully indexed {processed_count} items from pk {pk_offset}."
+                if not processed_count % 1000:
+                    # Log every 1000 parent documents processed.
+                    log_last_document_indexed(
+                        item_id, compose_redis_key(search_type, event_doc_type)
+                    )
+            self.stdout.write(
+                f"Successfully indexed {processed_count} items from pk {pk_offset}."
+            )
+
+    def index_documents_from_event_table(
+        self,
+        iterable: Iterable,
```

Comment:
Maybe call this something better: `items_to_update`, or even `items`?

Reply:
yeah, I renamed it to `events_to_update`, since this is directly related to events in history tables.